File: Reverse Words in a String.js


var reverseWords = function (s) {
  return s
    .trim()
    .split(" ")
    .filter((word) => word.length > 0)
    .reverse()
    .join(" ");
};



//////////////////////////////////////////////////////////////////

function reverseWords(s) {
  let words = [],
    word = "";
  for (let i = 0; i <= s.length; i++) {
    if (s[i] !== " " && i !== s.length) {
      word += s[i];
    } else if (word.length > 0) {
      words.push(word);
      word = "";
    }
  }
  return words.reverse().join(" ");
}

// Example usage
let example = "the sky is blue";
let output = reverseWords(example);
console.log(output);

File: Reverse Words in a String.py
def reverseWords(s: str) -> str:
    return ' '.join(reversed(s.split()))


# Example usage:
example = "the sky is blue"
output = reverseWords(example)



# SOLUTION 2 //////////////////////////////////////////////

def reverseWords(s: str) -> str:
    return ' '.join(s.split()[::-1])


# Example usage:
example = "the sky is blue"
output = reverseWords(example)

# SOLUTION 3     //////////////////////////////////////////////


def reverseWords(s: str) -> str:
    words = []
    length = len(s)
    i = 0

    while i < length:
        if s[i] != ' ':
            start = i
            while i < length and s[i] != ' ':
                i += 1
            words.append(s[start:i])
        i += 1

    return ' '.join(reversed(words))


# Example usage:
example = "the sky is blue"
output = reverseWords(example)
